lib/cli/github_settings/do_set_branch_protection.js

```diff
@@ -152,33 +163,86 @@ function do_set_branch_protection(subcmd, opts, args, cb) {
                         return;
                     }
 
-                    var reqOpts = {
-                        owner: 'joyent',
-                        repo: repoName,
-                        branch: 'master',
+                    var reqOpts = {};
+                    var lockMessage = '';
 
-                        enforce_admins: true,
-                        required_pull_request_reviews: {
-                            dismissal_restrictions: {},
-                            dismiss_stale_reviews: true,
-                            require_code_owner_reviews: false,
-                            required_approving_review_count: 1
-                        },
-                        restrictions: null,
-                        // Don't muck with this value yet, until MANTA-4598 is
-                        // decided and complete.
-                        required_status_checks: ctx.masterProt
-                            ? ctx.masterProt.required_status_checks
-                            : null
-                    };
+                    if (ctx.unlockUsers) {
+                        lockMessage =
+                            'Allowing only ' +
+                            ctx.unlockUsers.join(', ') +
+                            ' to push to "master" ' +
+                            'branch of %s ' +
+                            '(https://github.com/joyent/%s/settings/branches)';
+                        reqOpts = {
+                            owner: 'joyent',
+                            repo: repoName,
+                            branch: 'master',
+
+                            enforce_admins: true,
+                            required_pull_request_reviews: null,
+                            restrictions: {
+                                users: ctx.unlockUsers,
+                                teams: []
+                            },
+                            required_status_checks: null
+                        };
+                        // fml
```

Comment:
I don't think this comment is helpful. If you are referring to the long string of careful `this && this.that && this.that.otherthis && ...`  I believe there is a jsprim.pluck or similar that could be used to shorten this, if you like.

Reply:
Sorry, I was complaining there - that seemed pretty ugly, and I wasn't sure if that was really acceptable? It seems weird that the octokit api doesn't fully fill-in the object with the missing fields.

Reply:
Yah, it is a pain to carefully traverse an object structure that might not exist. Welcome to dynamic languages!